### PR TITLE
Fix the consul secret backends renewal revocation problem

### LINKED ISSUE #1433: Possible issue when renewing leases
Hi!

I was playing a bit with Vault (5.2) and Consul (6.4) and I found something weird.
I mounted the consul backend, configured the access and created a role to get tokens. To test token renewal I changed the default ttl lease to 30 seconds.

```bash
vault mount consul
vault write consul/config/access \
  address=myip:port \
  token=mytoken
echo 'key "docker/" { policy = "write" }' | base64 | vault write consul/roles/docker policy=- token_type=client
vault mount-tune -default-lease-ttl=30s consul
```
If I request a token, everything works fine:

```bash
vault read -format=json consul/creds/docker 
{
	"lease_id": "consul/creds/docker/2b3e5140-c1f3-4052-9a20-a7ec8931bf1e",
	"lease_duration": 30,
	"renewable": true,
	"data": {
		"token": "614c0705-4747-ffdf-70fb-7d224e36a18a"
	},
	"warnings": null
}
```
And after 30 seconds the token is removed in vault and deleted from consul, so everything works fine.
But if I try to renew the token, the response is the next one:
```bash
vault read -format=json consul/creds/docker 
{
	"lease_id": "consul/creds/docker/106b532f-c605-c6f6-4287-6fcc752a725e",
	"lease_duration": 30,
	"renewable": true,
	"data": {
		"token": "2b424dfc-b23c-8825-df39-450d696ae4f7"
	},
	"warnings": null
}
vault renew -format=json consul/creds/docker/106b532f-c605-c6f6-4287-6fcc752a725e 10
{
	"lease_id": "consul/creds/docker/106b532f-c605-c6f6-4287-6fcc752a725e",
	"lease_duration": 10,
	"renewable": true,
	"data": null,
	"warnings": null
}
```
The *lease_duration* that is received is 10 (the same as I sent as increment). Shouldn't it be remaining + increment?
Another weird thing is that if I try to renew the token after that 10 seconds, Vault sais that: `lease not found or lease is not renewable`. I guess that I should receive this after 40 seconds, not 10.
Lastly, if I make the renew, the token its never removed from consul (at least not after those 10 or 40 seconds). I check after half an hour and the token is still in consul. I'm able to use it. 

I don't know if this is a specific problem for consul secret backend or affects all of them. 

Maybe part of the problem is [here](https://github.com/hashicorp/vault/blob/master/logical/framework/lease.go#L81). 

Maybe this is the expected behaviour and I'm doing something wrong...

### PR DESCRIPTION
**Pull request by vishalnayak**
Did manual testing and it works fine.

Fixes #1433

### CODE REVIEW

File: builtin/logical/consul/secret_token.go [hunk 1]
```diff
@@ -37,7 +39,12 @@ func secretTokenRevoke(
...
 	}
 
-	_, err = c.ACL().Destroy(d.Get("token").(string), nil)
+	tokenRaw, ok := req.Secret.InternalData["token"]
+	if !ok {
+		return nil, fmt.Errorf("secret is missing internal data: token")
+	}
+
+	_, err = c.ACL().Destroy(tokenRaw.(string), nil)
 	if err != nil {
```
Comment (jefferai) [inline builtin/logical/consul/secret_token.go]: Should this then return a real error, not a logical.ErrorResponse, so the expiration manager sees that it wasn't successful?

Comment (jefferai) [inline builtin/logical/consul/secret_token.go]: Doh. I didn't look at the rest before writing this. :clap:

Comment (vishalnayak) [inline builtin/logical/consul/secret_token.go]: It is still an internal error. I changed it to `return nil, err`

**Comment by jefferai** [comment]
LGTM!

**Pull request merged**